test/unit/specs/core/createEventManager.spec.js

```diff
@@ -131,33 +136,24 @@ describe("createEventManager", () => {
       });
     });
 
-    it("calls onBeforeEvent before consent and onBeforeDataCollectionRequest after", () => {
-      const deferred = defer();
-      consent.awaitConsent = () => deferred.promise;
+    it("allows components and consent to pause the lifecycle", () => {
+      const onBeforeEventDeferred = defer();
+      const consentDeferred = defer();
+      lifecycle.onBeforeEvent.and.returnValue(onBeforeEventDeferred.promise);
+      consent.awaitConsent.and.returnValue(consentDeferred.promise);
       eventManager.sendEvent(event);
+      expect(lifecycle.onBeforeEvent).toHaveBeenCalled();
       return flushPromiseChains()
         .then(() => {
-          expect(lifecycle.onBeforeEvent).toHaveBeenCalled();
-          expect(
-            lifecycle.onBeforeDataCollectionRequest
```

Comment:
`onBeforeDataCollectionRequest` has been removed so I had to rework this a bit. I also improved it a bit to consider `awaitConsent`'s effect on the lifecycle.